### PR TITLE
systemd patches for Plymouth

### PR DESCRIPTION
**Pull request by abbradar**
###### Motivation for this change

This is a mass-rebuildy part of #16763 and also a small cleanup. It needs patches to systemd https://github.com/NixOS/systemd/pull/5 and https://github.com/NixOS/systemd/pull/6 -- in this PR I use my own fork but this _should not_ be merged as is.
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by abbradar** [comment]
Ouch, I've posted it against `master` by mistake. Given that most likely this would be merged by one who would also merge systemd commits and update revision I expect this to be merged by hand, so let's leave it as is. Ping me if you want this to be re-created anyway!

**Comment by abbradar** [comment]
Whoever considers this for merging please also merge https://github.com/NixOS/nixpkgs/pull/16642 to reduce mass rebuilds -- I consider it ready.

File: pkgs/os-specific/linux/systemd/default.nix [hunk 1]
```diff
@@ -14,14 +14,12 @@ stdenv.mkDerivation rec {
   name = "systemd-${version}";
 
   src = fetchFromGitHub {
-    owner = "NixOS";
+    owner = "abbradar";
```
Comment (edolstra) [inline pkgs/os-specific/linux/systemd/default.nix]: I've applied the NixOS/systemd patches, so this can be changed back.

**Comment by abbradar** [comment]
Closed in favor of #16797

**Pull request closed without merging**